openlibrary/plugins/upstream/account.py

```diff
@@ -869,6 +869,39 @@ def GET(self):
         entity = contributions if i.entity == 'Contribution' else contact
         return delegate.RawText(simplejson.dumps(entity), content_type="application/json")
 
+class import_books(delegate.page):
+    path = "/account/import"
+
+    @require_login
+    def GET(self):
+        return render['account/import']()
+
+class fetch_goodreads(delegate.page):
+    path = "/account/import/goodreads"
+
+    @require_login
+    def POST(self):
+        import csv
+        i = web.input(csv={})
+        csv_file = csv.reader((i.csv.value).splitlines(),
+                              delimiter=',', quotechar='"')
+        header = csv_file.next()
```

Comment:
If no csv file is uploaded, `csv_file.next()` will hit a `StopIteration` and raise an error. We likely need to deal with the case where no csv file is uploaded (e.g. deactivate the button until csv file uploaded)